File: 01--trebuchet/a.hs
#!/usr/bin/env runhaskell
-- Usage:
--   ./a.hs PATH_TO_INPUT_FILE

import Data.Char (isDigit)
import System.Environment (getArgs)

main = do
  [inputPath] <- getArgs
  text <- readFile inputPath
  print $ solve text

solve text =
  sum $ map getCalibrationValue (lines text)

getCalibrationValue line =
  read [firstDigit, lastDigit]
  where
    digits = filter isDigit line
    firstDigit = head digits
    lastDigit = last digits

File: 01--trebuchet/a.py
#!/usr/bin/env python3
'''
Usage:
    ./a.py PATH_TO_INPUT_FILE
'''

import string
import sys


def main():
    answer = 0
    for line in open(sys.argv[1]).read().splitlines():
        digits = [ch for ch in line if ch in string.digits]
        calibration_value = int(
            digits[0] +
            digits[-1]
        )
        answer += calibration_value
    print(answer)


if __name__ == '__main__':
    main()
